packages/protocol/StreamingScreenShotCache.ts

```diff
@@ -0,0 +1,102 @@
+import { ExecutionPoint, PauseId, ScreenShot, repaintGraphicsResult } from "@replayio/protocol";
+import { Cache, createCache, createStreamingCache } from "suspense";
+
+import { PaintsCache, findMostRecentPaint } from "protocol/PaintsCache";
+import { pauseIdCache } from "replay-next/src/suspense/PauseCache";
+import { screenshotCache } from "replay-next/src/suspense/ScreenshotCache";
+import { ReplayClientInterface } from "shared/client/types";
+
+export const RepaintGraphicsCache: Cache<
+  [replayClient: ReplayClientInterface, pauseId: PauseId],
+  repaintGraphicsResult | null
+> = createCache({
+  config: { immutable: true },
+  debugLabel: "RepaintGraphicsCache",
+  getKey: ([replayClient, pauseId]) => pauseId,
+  load: async ([replayClient, pauseId]) => {
+    return replayClient.repaintGraphics(pauseId);
+  },
+});
+
+type StreamingScreenShotCacheStatus =
+  | "complete"
+  | "loading-imprecise"
+  | "loading-precise"
+  | "loading-failed";
+
+export const StreamingScreenShotCache = createStreamingCache<
+  [replayClient: ReplayClientInterface, time: number, executionPoint: ExecutionPoint | null],
+  ScreenShot | undefined,
+  StreamingScreenShotCacheStatus
+>({
+  debugLabel: "StreamingScreenShotCache",
+  getKey: (replayClient, time, executionPoint) => `${time}:${executionPoint}`,
+  load: async ({ update, reject, resolve }, replayClient, time, executionPoint) => {
+    let screenShot: ScreenShot | undefined = undefined;
+
+    update(screenShot, 0, "loading-imprecise");
```

Comment:
:question: What is "imprecise" in this case?

Reply:
Imprecise means it's the nearest cached paint. We show it first because it's reliably and fast to fetch, then we ask for a (more precise) repaint below (which may be slow and/or may fail to load entirely).

Reply:
I struggled a bit to think of the right terminology to use here. I guess I could have used "requesting-cached-paint" and "requesting-repaint"

Do you think those would be clearer? I don't mind changing this. It's only referenced in one place.